adapters/mobilefuse/mobilefuse.go

```diff
@@ -0,0 +1,193 @@
+package mobilefuse
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strconv"
+	"text/template"
+)
+
+type MobilefuseAdapter struct {
+	EndpointTemplate template.Template
+}
+
+func NewMobilefuseBidder(endpointTemplate string) adapters.Bidder {
+	parsedTemplate, errs := template.New("endpointTemplate").Parse(endpointTemplate)
+
+	if errs != nil {
+		glog.Fatal("Unable parse endpoint template: " + errs.Error())
+		return nil
+	}
+
+	return &MobilefuseAdapter{EndpointTemplate: *parsedTemplate}
+}
+
+func (adapter *MobilefuseAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	adapterRequest, errs := adapter.makeRequest(request)
+
+	if errs == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+
+	return adapterRequests, errs
+}
+
+func (adapter *MobilefuseAdapter) MakeBids(incomingRequest *openrtb.BidRequest, outgoingRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var incomingBidResponse openrtb.BidResponse
+
+	if err := json.Unmarshal(response.Body, &incomingBidResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	outgoingBidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
```

Comment:
Do you typically expect a single bid? If not, consider raising the `1` to what you typically expect as a response to avoid memory re-allocation.

Reply:
Yes, we have a maximum of one bid per response.

Reply:
Ok, cool. Thank you for confirming. I'm good with this.